src/application/components/Layouts/SettingsModal.tsx

```diff
@@ -0,0 +1,77 @@
+import { Dispatch, Fragment, MutableRefObject, SetStateAction } from "react";
+import { Dialog, Transition } from "@headlessui/react";
+
+declare const VERSION: string;
+
+export function Modal({
+  open,
+  setOpen,
+  cancelButtonRef,
+}: {
+  open: boolean;
+  cancelButtonRef: MutableRefObject<null>;
+  setOpen: Dispatch<SetStateAction<boolean>>;
+}) {
+  return (
+    <Transition.Root show={open} as={Fragment}>
+      <Dialog
+        as="div"
+        className="relative z-10"
+        initialFocus={cancelButtonRef}
+        onClose={setOpen}
+      >
+        <Transition.Child
+          as={Fragment}
+          enter="ease-out duration-300"
+          enterFrom="opacity-0"
+          enterTo="opacity-100"
+          leave="ease-in duration-200"
+          leaveFrom="opacity-100"
+          leaveTo="opacity-0"
+        >
+          <div className="fixed inset-0 bg-grey-darker bg-opacity-75 transition-opacity" />
+        </Transition.Child>
+
+        <div className="fixed inset-0 z-10 w-screen overflow-y-auto">
+          <div className="flex min-h-full items-end justify-center p-4 text-center sm:items-center sm:p-0">
+            <Transition.Child
+              as={Fragment}
+              enter="ease-out duration-300"
+              enterFrom="opacity-0 translate-y-4 sm:translate-y-0 sm:scale-95"
+              enterTo="opacity-100 translate-y-0 sm:scale-100"
+              leave="ease-in duration-200"
+              leaveFrom="opacity-100 translate-y-0 sm:scale-100"
+              leaveTo="opacity-0 translate-y-4 sm:translate-y-0 sm:scale-95"
+            >
+              <Dialog.Panel className="relative transform overflow-hidden rounded-lg bg-white px-4 pb-4 pt-5 text-left shadow-xl transition-all sm:my-8 sm:w-full max-w-sm sm:p-6">
+                <div>
```

Comment:
I'm curious if this extra `div` is needed since you have another wrapping div as a child of this one. Could you consolidate into a single one?

In fact, the `Dialog.Panel` seems to do a pretty good job of wrapping the whole content, so I think you can remove both this `div` and the child `div` as well. This doesn't seem to change the content in any way if I remove both of these when testing locally.

Reply:
Addressed in https://github.com/apollographql/apollo-client-devtools/pull/1128/commits/41bdf86d24adcb2decbd996401584e8c7914dc99